### PR TITLE
Added support for vault/no-vault 

### PR DESCRIPTION
**Pull request by nabadger**
- Added required VAULT_ENABLED arg  (by not setting this, we can test locally)
- Tidy up script with sensible defaults/overides

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
+# Default settings (else take from env)
+DB_HOST=${DB_HOST:-127.0.0.1}
+DB_CONF=${DB_CONF:-$HOME/.my.cnf}
+VAULT_ENABLED=${VAULT_ENABLED:-0}
+KUBERNETES_SERVICE_HOST=${KUBERNETES_SERVICE_HOST:-}
 
+# Grab db-values from vault if vault is enabled and we are running
+# inside kubernetes.
+if [ -n "${KUBERNETES_SERVICE_HOST}" ] && [ "${VAULT_ENABLED}" = "1" ]; then
+  echo "in"
```
Comment (primeroz) [inline docker-entrypoint.sh:13]: is this a `debug echo` ?

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
   DB_NAME=$(vault-env env | grep DB_NAME | cut -d"=" -f2)
   DB_USER=$(vault-env env | grep DB_USER | cut -d"=" -f2)
   DB_PASSWORD=$(vault-env env | grep DB_PASSWORD | cut -d"=" -f2)
+fi
 
-cat << EOF > /home/mintel/.my.cnf
+cat << EOF > "$DB_CONF"
 [client]
 
 database=$DB_NAME
```
Comment (primeroz) [inline docker-entrypoint.sh:22]: whre does `DB_NAME` and `DB_USER` and `DB_PASSWORD` come from when vault is enabled ?

File: docker-entrypoint.sh [hunk 3]
```diff
@@ -1,22 +1,28 @@
 #!/bin/sh
+set -e -o nounset
 
-set -e
-
-if [ -n "${KUBERNETES_SERVICE_HOST}" ]; then
+# Default settings (else take from env)
+DB_HOST=${DB_HOST:-127.0.0.1}
+DB_CONF=${DB_CONF:-$HOME/.my.cnf}
+VAULT_ENABLED=${VAULT_ENABLED:-0}
```
Comment (primeroz) [inline docker-entrypoint.sh:7]: I would use a `false` or `true` string

Comment (primeroz) [reply docker-entrypoint.sh:7]: or like for `KUBERNETES_SERVICE_HOST` just default to empty and whatever its value is `true`

File: docker-entrypoint.sh (hunk 2 above)
Comment (nabadger) [reply docker-entrypoint.sh:22]: Comes from vault when its enabled, else upto the user

```
  DB_NAME=$(vault-env env | grep DB_NAME | cut -d"=" -f2)
  DB_USER=$(vault-env env | grep DB_USER | cut -d"=" -f2)
  DB_PASSWORD=$(vault-env env | grep DB_PASSWORD | cut -d"=" -f2)
```

File: docker-entrypoint.sh (hunk 1 above)
Comment (nabadger) [reply docker-entrypoint.sh]: My bad, deleted

File: docker-entrypoint.sh (hunk 3 above)
Comment (nabadger) [reply docker-entrypoint.sh]: Using true/false

File: docker-entrypoint.sh (hunk 2 above)
Comment (primeroz) [reply docker-entrypoint.sh:21]: > else upto the user

yeah that was my point. 

Should you error out if `is not vault` and `users did not define those` ?  Is not good practice to expect VARS but have no defaults and not error out if they are not defined

Comment (nabadger) [reply docker-entrypoint.sh:21]: It will error out because of `-o nounset`

Comment (primeroz) [reply docker-entrypoint.sh:21]: ok i did not see that

**Review by primeroz — APPROVED** [review]

**Pull request merged**